Mage.Sets/src/mage/cards/c/CustodyBattle.java

```diff
@@ -0,0 +1,176 @@
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author TheElk801
+ */
+public class CustodyBattle extends CardImpl {
+
+    private static final FilterControlledPermanent filter = new FilterControlledLandPermanent();
+
+    public CustodyBattle(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{R}");
+
+        this.subtype.add("Aura");
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        Ability ability = new EnchantAbility(auraTarget.getTargetName());
+        this.addAbility(ability);
+
+        // Enchanted creature has "At the beginning of your upkeep, target opponent gains control of this creature unless you sacrifice a land."
+        ability = new BeginningOfUpkeepTriggeredAbility(new CustodyBattleUnlessPaysEffect(new SacrificeTargetCost(new TargetControlledPermanent(filter))), TargetController.YOU, false);
+        ability.addTarget(new TargetOpponent());
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityAttachedEffect(ability, AttachmentType.AURA)));
+
+    }
+
+    public CustodyBattle(final CustodyBattle card) {
+        super(card);
+    }
+
+    @Override
+    public CustodyBattle copy() {
+        return new CustodyBattle(this);
+    }
+}
+
+class GiveControlEffect extends ContinuousEffectImpl {
+
+    public GiveControlEffect() {
+        super(Duration.Custom, Layer.ControlChangingEffects_2, SubLayer.NA, Outcome.GainControl);
+        staticText = "Target opponent gains control of {this}";
+    }
+
+    public GiveControlEffect(final GiveControlEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GiveControlEffect copy() {
+        return new GiveControlEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = (Permanent) source.getSourceObjectIfItStillExists(game);
+        if (permanent != null) {
+            return permanent.changeControllerId(source.getFirstTarget(), game);
+        } else {
+            discard();
+        }
+        return false;
+    }
+
+}
+
+class CustodyBattleUnlessPaysEffect extends OneShotEffect {
+
+    protected Cost cost;
+
+    public CustodyBattleUnlessPaysEffect(Cost cost) {
+        super(Outcome.Sacrifice);
+        staticText = "target opponent gains control of {this} unless you sacrifice a land";
+        this.cost = cost;
+    }
+
+    public CustodyBattleUnlessPaysEffect(final CustodyBattleUnlessPaysEffect effect) {
+        super(effect);
+        this.cost = effect.cost.copy();
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
```

Comment:
use getPermanentorLKI otherwisw thr resolution will fail if the Enchantment left the battlefield between triggering and resolution.